### PR TITLE
Propagate the query param when navigating

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Fixes https://github.com/web-platform-tests/wpt.fyi/issues/563

The query param is not part of the test-runs query, but a (separate) filtering query. Additionally, you can't actually override the `computed` field (on a property) value, so we can't branch what constitutes the query.

### CODE REVIEW

File: webapp/components/test-search.html [hunk 1]
```diff
@@ -49,8 +49,6 @@
             type: String,
             value: 'Search test files, like \'cors/allow-headers.htm\', then press <Enter>',
           },
-          items: Array,
-          itemKey: String,
```
Comment (lukebjerring) [inline webapp/components/test-search.html]: These fields are unused, and mislead me in the binding setup.

File: webapp/components/wpt-results.html [hunk 2]
```diff
@@ -378,28 +375,29 @@ <h3>
           .filter(name => name.startsWith(path));
       }
 
-      computeSearchURL(testRuns, search) {
-        let url = '/api/search?';
+      computeSearchURL(testRuns, query) {
+        let url = new URL('/api/search', window.location);
```
Comment (lukebjerring) [inline webapp/components/wpt-results.html]: This is a drive-by cleanup.

File: webapp/components/interop.html [hunk 3]
```diff
@@ -214,11 +214,12 @@
...
-  /* global TestRunsBase, SelfNavigation, LoadingState */
-  class WPTInterop extends SelfNavigation(LoadingState(TestRunsBase)) {
-    static get is() {
-      return 'wpt-interop';
-    }
+  /* global TestRunsBase, SelfNavigation, LoadingState, ResultsNavigation */
+  class WPTInterop extends ResultsNavigation(
+    SelfNavigation(LoadingState(TestRunsBase)),
+    'interopQueryParams(sha, labels, productSpecs, maxCount, search)') {
+    static get is() { return 'wpt-interop'; }
```
Comment (mdittmer) [inline webapp/components/interop.html]: Does our linter like this one-liner?

File: webapp/components/interop.html [hunk 4]
```diff
@@ -214,11 +214,12 @@
...
 
-  /* global TestRunsBase, SelfNavigation, LoadingState */
-  class WPTInterop extends SelfNavigation(LoadingState(TestRunsBase)) {
-    static get is() {
-      return 'wpt-interop';
-    }
+  /* global TestRunsBase, SelfNavigation, LoadingState, ResultsNavigation */
+  class WPTInterop extends ResultsNavigation(
+    SelfNavigation(LoadingState(TestRunsBase)),
+    'interopQueryParams(sha, labels, productSpecs, maxCount, search)') {
```
Comment (mdittmer) [inline webapp/components/interop.html]: This is starting to feel like C++ template coding horror :scream:

File: webapp/components/interop.html [hunk 5]
```diff
@@ -262,6 +263,9 @@
         // passRateMetadata contains the url for the JSON blob of allMetrics;
         // both fetches need to succeed + parse.
         this.interopLoadFailed = !(this.passRateMetadata && this.allMetrics);
+        if (!this.interopLoadFailed && this.search) {
+          this.shadowRoot.querySelector('test-search').commitQuery();
```
Comment (mdittmer) [inline webapp/components/interop.html]: This feels wrong. The desired effect is that _this_ component runs application logic associated with having both interop data and a query. 

I would prefer to write a helper for `handleCommitQuery(e)`, like `onQuery(q)`; `handleCommitQuery` can delegate to `onQuery` after extracting `q`, and this line of code becomes `this.onQuery(this.search)`.

File: webdriver/search_test.go [hunk 6]
```diff
@@ -24,58 +32,62 @@ func TestSearch(t *testing.T) {
...
+	var pathParts []selenium.WebElement
+	var err error
 	filteredPathPartsCondition := func(wd selenium.WebDriver) (bool, error) {
-		pathParts, err := getPathPartElements(wd, elementName)
+		pathParts, err = getPathPartElements(wd, elementName)
 		if err != nil {
 			return false, err
 		}
-		return len(pathParts) > 0 && len(pathParts) < numInitialPathParts, nil
+		return len(pathParts) == 1, nil
```
Comment (mdittmer) [inline webdriver/search_test.go]: Does this test use static input data? If not, the introduction of something in another folder named `2dcontext` could cause an unexpected breakage.

File: webapp/components/interop.html (hunk 5 above)
Comment (lukebjerring) [reply webapp/components/interop.html]: re-jigged `handleSearchCommit` to take just the query. PTAL.

File: webdriver/search_test.go (hunk 6 above)
Comment (lukebjerring) [reply webdriver/search_test.go]: It does locally, but does not in staging. Been this way a long time.. I'll leave it for you to have the chance to say "told you so" later ;)
